cmd/restic/cmd_check_test.go

```diff
@@ -163,3 +165,75 @@ func TestSelectNoRandomPacksByFileSize(t *testing.T) {
 	selectedPacks := selectRandomPacksByFileSize(testPacks, 10, 500)
 	rtest.Assert(t, len(selectedPacks) == 0, "Expected 0 selected packs")
 }
+
+func TestPrepareCheckCache(t *testing.T) {
+	// Create a temporary directory for the cache
+	tmpDirBase, err := os.MkdirTemp("", "test-prepare-check-cache-basepath")
+	if err != nil {
+		t.Fatalf("Failed to create temporary directory: %v", err)
+	}
+	defer os.RemoveAll(tmpDirBase)
+
+	testCases := []struct {
+		opts        CheckOptions
+		cacheExists bool
+		expectNoErr bool
+	}{
+		{CheckOptions{WithCache: true}, false, true}, // Shouldn't create temp directory
+		{CheckOptions{WithCache: false}, true, true}, // Should create temp directory
+	}
+
+	for _, testCase := range testCases {
+		t.Run("", func(t *testing.T) {
+			gopts := GlobalOptions{CacheDir: tmpDirBase}
+			cleanup := prepareCheckCache(testCase.opts, &gopts)
+
+			if !testCase.opts.WithCache {
+				// If using a temporary cache directory, the cache directory should exist
+				// listing all directories inside tmpDirBase (cacheDir)
+				// one directory should be tmpDir created by prepareCheckCache with 'restic-check-cache-' in path
+				files, err := os.ReadDir(tmpDirBase)
```

Comment:
Both code paths call `os.ReadDir`, that call should be moved out of the if block.